src/oop/static_modifier/medium/q2/MainTest.java

```diff
@@ -0,0 +1,138 @@
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    private ObjectTest classInstance;
+    private ObjectTest testInstance;
+    private final String objectName = "Laptop";
+    private final String testName = "TestObject";
+    private final String intAttribute1 = "identification";
+    private final String intAttribute2 = "initial";
+    private final String getMethodName = "getIdentification";
+    private final String setMethodName = "setIdentification";
+    private final String methodName = "incrementer";
+    private final int initialStart = 0;
+    private final int incrementAmount = 5;
+
+
+    @BeforeEach
+    public void setup() {
+        String classPath = "oop.static_modifier.medium.q2." + objectName;
+        classInstance = new ObjectTest(classPath);
+        String testPath = "oop.static_modifier.medium.q2." + testName;
+        testInstance = new ObjectTest(testPath);
+    }
+
+    @Test
+    public void laptopClassHasRequiredAttributes() {
+        assertTrue(classInstance.hasField(intAttribute1),
+                String.format("Your %s class is missing the %s attribute. Please make sure that is is spelt correctly.", objectName, intAttribute1));
+        assertTrue(classInstance.hasField(intAttribute2),
+                String.format("Your %s class is missing the %s attribute. Please make sure that is is spelt correctly.", objectName, intAttribute2));
+        assertTrue(classInstance.hasField(intAttribute1, int.class),
+                String.format("Your %s attribute does not have the correct type.", intAttribute1));
+        assertTrue(classInstance.hasField(intAttribute2, int.class),
+                String.format("Your %s attribute does not have the correct type.", intAttribute2));
+        assertTrue(classInstance.hasModifier(intAttribute1, "private"),
+                String.format("Your %s attribute does not have the correct visibility modifier.", intAttribute1));
+        assertTrue(classInstance.hasModifier(intAttribute2, "private"),
+                String.format("Your %s attribute does not have the correct visibility modifier.", intAttribute2));
+        assertTrue(classInstance.hasModifier(intAttribute2, "static"),
+                String.format("Your %s attribute is missing the static modifier.", intAttribute2));
+    }
+
+    @RepeatedTest(3)
+    public void laptopClassHasCorrectConstructor(RepetitionInfo repetitionInfo) throws Throwable {
+        if (repetitionInfo.getCurrentRepetition() == 1) {
+            classInstance.setFieldValue(null, initialStart, intAttribute2);
```

Comment:
We should not set this here as we want to make sure that the student is setting the initial value of this attribute to zero. You may have to order the tests so this one runs before some of the other ones. If you are not sure how to do that, take a look at #694.